### PR TITLE
[KIECLOUD-646] - Rebranding changes on the kie-cloud-operator repository

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

### CODE REVIEW

**Review by jakubschwan — CHANGES_REQUESTED** [review]
Changes requested.
We should avoid to any changes in kieapp crd. This CRD will be shared by 2 operators.
The operator for bamoe should have different naming to not break the existing operators.
The clusterservice version should also have newer version set for bamoe. It is also expected that csv file for the first release of bamoe does not contains `replaces` property.

**Comment by spolti** [comment]
@jakubschwan 
I think it makes sense not update the crd.
Let's wait to see what changes will be required to move it further.

**Review by Ginxo — APPROVED** [review]

**Review by desmax74 — APPROVED** [review]

File: pkg/controller/kieapp/constants/constants.go [hunk 1]
```diff
@@ -73,12 +73,18 @@ const (
 	LabelRHsubcomponentType = "rht.subcomp_t"
 	// LabelRHcompany used as metering label
 	LabelRHcompany = "com.company"
+	// IBMBamoeImageContext IBM image context
+	IBMBamoeImageContext = "ibm-bamoe"
+	// IBMBamoeImagePrefix IBM image prefix
+	IBMBamoeImagePrefix = "bamoe"
 	// RhpamPrefix RHPAM prefix
 	RhpamPrefix = "rhpam"
```
Comment (desmax74) [inline pkg/controller/kieapp/constants/constants.go:81]: Is still needed ?

File: pkg/controller/kieapp/constants/constants.go [hunk 2]
```diff
@@ -73,12 +73,18 @@ const (
...
 	// RhdmPrefix RHDM prefix
 	RhdmPrefix = "rhdm"
```
Comment (desmax74) [inline pkg/controller/kieapp/constants/constants.go:83]: Is still needed ?

File: pkg/controller/kieapp/defaults/defaults.go [hunk 3]
```diff
@@ -212,19 +212,15 @@ func setProductLabels(cr *api.KieApp, env *api.Environment) {
 func getConsoleName(cr *api.KieApp) string {
 	env := string(cr.Spec.Environment)
 	consoleName := constants.RhpamBusinessCentral
-	if strings.Contains(env, constants.RhdmPrefix) && !isGE713(cr) {
-		consoleName = constants.RhdmDecisionCentral
-	} else if strings.Contains(env, constants.RhpamPrefix) && strings.Contains(env, constants.Production) {
+	if strings.Contains(env, constants.RhpamPrefix) && strings.Contains(env, constants.Production) {
```
Comment (desmax74) [inline pkg/controller/kieapp/defaults/defaults.go:215]: is still valid the RhpamPrefix with Ithe presence of BMBamoeImagePrefix ?

File: pkg/controller/kieapp/defaults/defaults.go [hunk 4]
```diff
@@ -444,9 +434,8 @@ func getConsoleTemplate(cr *api.KieApp) api.ConsoleTemplate {
 		}
 		template.Replicas = *cr.Status.Applied.Objects.Console.Replicas
 		template.Name = envConstants.App.Prefix
-		cMajor, _, _ := GetMajorMinorMicro(cr.Status.Applied.Version)
 
-		template.ImageURL = constants.ImageRegistry + "/" + product + "-" + cMajor + "/" + product + "-" + envConstants.App.ImageName + constants.RhelVersion + ":" + cr.Status.Applied.Version
+		template.ImageURL = constants.ConnectImageRegistry + "/" + constants.IBMBamoeImageContext + "/" + constants.IBMBamoeImagePrefix + "-" + envConstants.App.ImageName + constants.RhelVersion + ":" + cr.Status.Applied.Version
```
Comment (desmax74) [inline pkg/controller/kieapp/defaults/defaults.go:438]: No version with BAMOE ?

File: pkg/controller/kieapp/constants/constants.go (hunk 2 above)
Comment (spolti) [reply pkg/controller/kieapp/constants/constants.go:83]: yes, due the environments, rhdm for decisions and rhpam for process.
better not to rename it at this time.

File: pkg/controller/kieapp/constants/constants.go (hunk 1 above)
Comment (spolti) [reply pkg/controller/kieapp/constants/constants.go:81]: Yes, it is tied to a few things, image name, env prefix e a few other things.

File: pkg/controller/kieapp/defaults/defaults.go (hunk 3 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults.go:215]: yes, the IBMBamoe is just for image names.

File: pkg/controller/kieapp/defaults/defaults.go (hunk 4 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults.go:438]: what do you mean?

**Review by sutaakar — APPROVED** [review]

File: deploy/operator.yaml [hunk 5]
```diff
@@ -37,59 +37,53 @@ spec:
...
+        - name: RELATED_IMAGE_DATAGRID_IMAGE_8.0.0
+          value: registry.redhat.io/datagrid/datagrid-8-rhel8:1.1
+        - name: RELATED_IMAGE_BROKER_IMAGE_8.0.0
+          value: registry.redhat.io/amq7/amq-broker-rhel8:7.9
         - name: RELATED_IMAGE_PAM_KIESERVER_IMAGE_7.13.0
-          value: registry.redhat.io/rhpam-7/rhpam-kieserver-rhel8:7.13.0
+          value: registry.connect.redhat.com/ibm-bamoe/bamoe-kieserver-rhel8:7.13.0
         - name: RELATED_IMAGE_PAM_CONTROLLER_IMAGE_7.13.0
-          value: registry.redhat.io/rhpam-7/rhpam-controller-rhel8:7.13.0
+          value: registry.connect.redhat.com/ibm-bamoe/bamoe-controller-rhel8:7.13.0
```
Comment (davidesalerno) [inline deploy/operator.yaml:67]: It is not clear to me why here we are using tag 7.13 and not 8.0.0 but it is probably right the one used.

Comment (jakubschwan) [reply deploy/operator.yaml:67]: Probably all related images for 7.13 should be removed. These images does not exists.

File: pkg/controller/kieapp/constants/constants.go [hunk 6]
```diff
@@ -12,9 +12,9 @@ var Ocp4Versions = []string{"4.10", "4.9", "4.8", "4.7", "4.6"}
 
 const (
 	// CurrentVersion product version supported
-	CurrentVersion = "7.13.0"
+	CurrentVersion = "8.0.0"
 	// PriorVersion product version supported
-	PriorVersion = "7.12.1"
+	PriorVersion = "7.13.0"
```
[the remark this answers is not in the document]
Comment (spolti) [reply pkg/controller/kieapp/constants/constants.go:17]: As we discussed, let's merge it as it is and remove the version part on a upcoming pull request.

File: pkg/controller/kieapp/constants/constants.go [hunk 7]
```diff
@@ -275,12 +274,10 @@ const (
...
 	//ImageNames for metering labels
-	RhpamSmartRouterImageName = RhpamPrefix + "-smartrouter-" + RhelVersion
-	RhpamControllerImageName  = RhpamPrefix + "-controller-" + RhelVersion
-	RhdmSmartRouterImageName  = RhdmPrefix + "-smartrouter-" + RhelVersion
-	RhdmControllerImageName   = RhdmPrefix + "-controller-" + RhelVersion
+	RhpamSmartRouterImageName = IBMBamoeImagePrefix + "-smartrouter-" + RhelVersion
+	RhpamControllerImageName  = IBMBamoeImagePrefix + "-controller-" + RhelVersion
 
-	RhdmDecisionCentral     = "decisioncentral"
+	//RhdmDecisionCentral     = "decisioncentral"
```
[the remark this answers is not in the document]
Comment (spolti) [reply pkg/controller/kieapp/constants/constants.go:280]: yes.

File: pkg/controller/kieapp/constants/constants.go [hunk 8]
```diff
@@ -373,64 +370,46 @@ var ProcessMigrationRequests = map[string]string{
...
+		Registry:  ConnectImageRegistry,
 		Context:   PamContext + "process-migration" + RhelVersion,
 	},
 	{
 		Var:       PamDashbuilderVar,
 		Component: RhpamPrefix + "-7-dashbuilder-rhel8-container",
-		Registry:  ImageRegistry,
+		Registry:  ConnectImageRegistry,
 		Context:   PamContext + "dashbuilder" + RhelVersion,
 	},
```
[the remark this answers is not in the document]
Comment (spolti) [reply pkg/controller/kieapp/constants/constants.go:414]: These components does not needs to be changed at this moment as it refers to the image component name on osbs.
not sure why it is used on operator though.

**Review by vaibhavjainwiz — COMMENTED** [review]
lgtm

**Pull request merged**